pkg/api/validation/validation.go

```diff
@@ -2865,6 +2865,19 @@ func ValidateService(service *api.Service) field.ErrorList {
 		nodePorts[key] = true
 	}
 
+	// Check for duplicate TargetPort
+	portsPath = specPath.Child("ports")
+	targetPorts := make(map[api.ServicePort]bool)
```

Comment:
This should just be `map[intstr.IntOrString]bool` for obviousness reasons. or `struct{}` instead of bool.